packages/logrotate/extra/delete-oldest-unmanaged-files.py

```diff
@@ -0,0 +1,52 @@
+#!/opt/mesosphere/bin/python3
+
+import logging
+import os
+import sys
+
+
+def main(directory, max_files, managed_file):
+    '''
+    Finds all files under the given `directory` and checks if the number
+    of files exceeds the `max_files`.  If so, deletes files starting from
+    the oldest (except for the `managed_file`) until the `max_files` is
+    no longer exceeded.
+
+    @type directory: str, absolute path of the directory to clean up
+    @type max_files: int, maximum number of files to keep
+    @type managed_file: str, one file (i.e. leading log) to excempt from cleanup
+    '''
+    logging.basicConfig(format='%(levelname)-4s ] %(message)s',
+                        level=logging.INFO)
+
+    # For simplicity, convert all paths to absolute paths.
+    directory = os.path.abspath(directory)
+    managed_file = os.path.abspath(managed_file)
+
+    # Build a list of all files.
+    # TODO(josephw): Do we want to delete directories too?
```

Comment:
The previous `postrotate` script would include directories by virtue of not filtering them out (i.e. find . -type f`).  Perhaps there is a degenerate case where logrotate segfaults due to having millions of directories under the log directory...